compat/win32/pthread.c

```diff
@@ -57,6 +57,8 @@ pthread_t pthread_self(void)
 	return t;
 }
 
+#ifdef GIT_WIN_XP_SUPPORT
```

Comment:
@dscho It has been about a month since you made a pea in Google group. Should I just remove the old code and define guard completely?

Maybe you want to wait for the next major 2.x release before merging this?